src/components/Room/Room.tsx

```diff
@@ -95,6 +129,26 @@ export function Room({
     setIsMessageSending(false)
   }
 
+  const addUserIfNotPresent = (userToAdd: User) => {
+    const userIndex = shellContext.peerList.findIndex(
+      peer => peer.peerId === userToAdd.peerId
+    )
+    if (userIndex === -1) {
+      shellContext.setPeerList([
+        ...shellContext.peerList,
+        { peerId: userToAdd.peerId, userId: userToAdd.userId },
+      ])
+    } else {
+      const peerListClone = [...shellContext.peerList]
+      peerListClone[userIndex].userId = userToAdd.userId
+      shellContext.setPeerList(peerListClone)
+    }
+  }
+
+  receiveUserId((userId: string, peerId?: string) => {
```

Comment:
There is no such scenario for now because we are always sending peerId => `await sendPeerId(userId, peerId)`.

But using makeAction from trystero, we have that for receiveing function:
`
export interface ActionReceiver<T> 
{
  (receiver: (data: T, peerId?: string, metadata?: Metadata) => void): void
}
`

Reply:
Ah I see. This seems like a potentially incorrect type definition in Trystero (which I wrote the first version of, so this might be my fault 😅). If that's the case, fixing this would be a larger task that we don't need to deal with in this PR. Thank you for getting to the bottom of this!